CRM/Core/Payment/BaseIPN.php

```diff
@@ -107,9 +107,38 @@ public function validateData(&$input, &$ids, &$objects, $required = TRUE, $payme
     $contact = new CRM_Contact_BAO_Contact();
     $contact->id = $ids['contact'];
     if (!$contact->find(TRUE)) {
-      CRM_Core_Error::debug_log_message("Could not find contact record: {$ids['contact']} in IPN request: " . print_r($input, TRUE));
-      echo "Failure: Could not find contact record: {$ids['contact']}<p>";
-      return FALSE;
+      // dev/core#1249 If the supplied contact_id cannot be found lets see if we can find a relevant one in the contribution Recur object or in the contribution object.
```

Comment:
I'd suggest putting in the comments the reason why the contact ID might not be there (e.g. the contact has since been merged).